### DISCUSSION TITLE
How can I preserve leading white space (indents) in Azure Data Studio XML output

### DISCUSSION THREAD

**Discussion by OneEyedSi** [microsoft/azuredatastudio#25614] [Q&A]
By default Azure Data Studio seems to remove leading whitespace from the inner text in XML columns in the results of a query. Is there any way to preserve the leading whitespace in XML inner text?

In contrast SQL Server Management Studio retains leading whitespace in the inner text when displaying XML.

Example:

```
declare @xml xml = 
'<root>
    <text>
Unindented text.
    Single indented text.
        Double-indented text.
    </text>
</root>';

select @xml;
```

Result in Azure Data Studio, after clicking on the XML output column in the result pane to open the XML in a separate window:

```
<root>
	<text>
		Unindented text.
		Single indented text.
		Double-indented text.
	</text>
</root>
```

Desired result:

```
<root>
	<text>
Unindented text.
	Single indented text.
		Double-indented text.
	</text>
</root>
```

I've tried unchecking setting `Query Editor > Results > Save as Xml: Formatted` but that doesn't make any difference so I assume it's for saving the results to a file.

**Comment by erinstellato-ms** [comment]
@OneEyedSi Correct in that the setting is only for saving.  If this is an option you'd like to see in the results pane, please create an issue (https://github.com/microsoft/azuredatastudio/issues/).  Thank you!
